### DISCUSSION TITLE
Looking at Shiny

### DISCUSSION THREAD

**Discussion by sej69** [shinyorg/shiny#1086] [Q&A]
Hello, I'm looking at Shiny for a project we are starting to work on.  I downloaded the sample code in VS 2022 but it doesn't compile and I'm in Nuget hell (it won't let me update/consolidate packages).  Do you have an update for the samples available?

**Comment by aritchie** [comment]
Did you read the readme on the sample repo?  The samples are setup to also compile with a submodule for debugging purposes, if the submodule is missing, the compile will complain but the sample itself will build & run just fine - which is specifically mentioned in the readme

What sample(s) are giving you the issue?
You shouldn't have to touch the nuget version.  All of the samples are setup against what version they were built with.

**Comment by sej69** [comment]
Specifically, I’m working with the SpeechRecognition.

I wiped the samples and grabbed the repository again just to be sure I’m running the GIT version.

I did read the readme, but I read it again.  I see why I cannot update the packages in Nuget now.  However, I can’t restore them either.  This is a sample of what I’m seeing doing a dotnet restore:

PM> dotnet restore
D:\Dev\Shiny\SpeechRecognition\Sample.UWP\Sample.UWP.csproj : warning NU1503: Skipping restore for project 'D:\Dev\Shiny\SpeechRecognition\Sample.UWP\Sample.UWP.csproj'. The project file may be invalid or missing targets required for restore. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample.Android\Sample.Android.csproj : warning NU1503: Skipping restore for project 'D:\Dev\Shiny\SpeechRecognition\Sample.Android\Sample.Android.csproj'. The project file may be invalid or missing targets required for restore. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample.iOS\Sample.iOS.csproj : warning NU1503: Skipping restore for project 'D:\Dev\Shiny\SpeechRecognition\Sample.iOS\Sample.iOS.csproj'. The project file may be invalid or missing targets required for restore. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample.UWP.Background\Sample.UWP.Background.csproj : warning NU1503: Skipping restore for project 'D:\Dev\Shiny\SpeechRecognition\Sample.UWP.Background\Sample.UWP.Background.csproj'. The project file may be invalid or missing targets required for restore. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
  Determining projects to restore...
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1504: Duplicate 'PackageReference' items found. Remove the duplicate items or use the Update functionality to ensure a consistent restore behavior. The duplicate 'PackageReference' items are: sqlite-net-pcl , sqlite-net-pcl 1.8.116; Xamarin.Forms , Xamarin.Forms 5.0.0.2401.
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1604: Project dependency Shiny.Core does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1604: Project dependency sqlite-net-pcl does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1604: Project dependency Xamarin.Forms does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1701: Package 'sqlite-net-pcl 1.0.10' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1701: Package 'SQLitePCL.raw_basic 0.7.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\_Template\Sample.Shared\Sample.Shared.csproj : warning NU1701: Package 'Xamarin.Forms 1.0.6186' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : warning NU1602: Sample.Shared does not provide an inclusive lower bound for dependency Shiny.Core. An approximate best match of Shiny.Core 1.0.0.1174 was resolved. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : warning NU1602: Sample.Shared does not provide an inclusive lower bound for dependency sqlite-net-pcl. An approximate best match of sqlite-net-pcl 1.0.10 was resolved. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : warning NU1701: Package 'sqlite-net-pcl 1.0.10' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : warning NU1701: Package 'SQLitePCL.raw_basic 0.7.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605: Detected package downgrade: Xamarin.Essentials from 1.7.1 to 1.7.0. Reference the package directly from the project to select a different version.  [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605:  Sample -> Sample.Shared -> Xamarin.Essentials (>= 1.7.1)  [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605:  Sample -> Xamarin.Essentials (>= 1.7.0) [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605: Detected package downgrade: Shiny from 2.5.0-preview-3108 to 2.3.0.2919-preview. Reference the package directly from the project to select a different version.  [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605:  Sample -> Sample.Shared -> Shiny (>= 2.5.0-preview-3108)  [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj : error NU1605:  Sample -> Shiny (>= 2.3.0.2919-preview) [D:\Dev\Shiny\SpeechRecognition\Speech.sln]
  Failed to restore D:\Dev\Shiny\SpeechRecognition\Sample\Sample.csproj (in 190 ms).

I love nuget, until I run into these issues…  I’d like to play around with this to figure out this will work for what I need before I invest a lot of time creating an app around it.

The package will not compile as the referenced packages are not there.

Any ideas?

**Reply by aritchie** [reply]
I'll update the speech recognition sample this week, but there is nothing from stopping you to use the samples as a guide which is their main intent.

**Comment by sej69** [comment]
Great thanks, Allan.

Yes, I know.  I’ve just got so much code to write and decisions to make on the next project it’s just easiest to use something that exists vs creating / learning to discover if something is going to work or not.  If I can get a few open hours in the next few days, I’ll probably do that if I don’t see an update posted.

Thanks again.
